### PR TITLE
Go term cosh

### LINKED ISSUE #3082: [Term Entry] Go Math `Cosh()`
### Reason/inspiration (optional)

We would like a new term entry for [Go](https://www.codecademy.com/resources/docs/go) about the `Cosh()` function.

The entry should go in a new file in `docs/content/go/concepts/math-functions/terms/cosh/cosh.md`

The entry should 

- Have a syntax section illustrating the function with a ` ```pseudo ` code block. 
- An example showing the function in use with a ` ```go ` code block. 
- A runnable codebyte example in a ` ```codebyte/golang` block. 

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by abu-hasib**
### Description
- This PR adds information about Go Math Cosh function. It closes #3082.
- Ran formating and ran all tests.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+## Syntax
+
+```pseudo
+import "math"
+
+CoshX = math.Cosh(x)
+```
+
+Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always return the hyperbolic consine value except some special cases:
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:26]:
```suggestion
Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always returns the hyperbolic cosine value except for some special cases:
```

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+- CoshX of `Cosh(±0)` is `1`
+- CoshX of `Cosh(±Inf)` is `+Inf`
+- CoshX of `Cosh(NaN)` is `NaN`
+
+## Example
+
+The following Golang Program calculates the hyperbolic cosine of `x` and prints out the result:
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:34]:
```suggestion
The following Golang program calculates the hyperbolic cosine of `x` and prints out the result:
```

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+The output will be:
+
+```shell
+74.21
+74.21
+```
+
+## Codebyte Example
+
+The following example shows how the `Cosh()` function handles float64 arguments including _special cases_ arguments.
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:59]:
```suggestion
The following example shows how the `Cosh()` function handles float64 arguments including special cases.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@abu-hasib I did a first review and left a few comments.

**Comment by abu-hasib** [comment]
@SSwiniarski I have gone through the review and resolved the comments.

**Review by SSwiniarski — APPROVED** [review]
@abu-hasib Looks good for a second review.

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## Syntax
+
+```pseudo
+import "math"
+
+CoshX = math.Cosh(x)
+```
+
+Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always returns the hyperbolic cosine value except for some special cases:
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:26]:
```suggestion
Where `CoshX` is the hyperbolic cosine value of `x`, it is of type `float64`. It always returns the hyperbolic cosine value except for some special cases:
```

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+- CoshX of `Cosh(±0)` is `1`
+- CoshX of `Cosh(±Inf)` is `+Inf`
+- CoshX of `Cosh(NaN)` is `NaN`
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:30]:
```suggestion
- `Cosh(±0)` is `1`
- `Cosh(±Inf)` is `+Inf`
- `Cosh(NaN)` is `NaN`
```

File: content/go/concepts/math-functions/terms/cosh/cosh.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+```shell
+74.21
+74.21
+```
+
+## Codebyte Example
+
+The following example shows how the `Cosh()` function handles float64 arguments including special cases.
+
+```codebyte/golang
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/cosh/cosh.md:61]: 2 of 3 of these results are duplicates from the previous example. Codebytes should not duplicate previous work. Ideally it should demonstrate some new functionality, information, etc. I would suggest updating the values or deleting the example.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@abu-hasib I've added a few additional comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by abu-hasib** [comment]
@caupolicandiaz resolved the comments

**Review by caupolicandiaz — APPROVED** [review]
@abu-hasib @SSwiniarski Looks good to merge

**Pull request merged**